tests/test_cases/test_cocotb/test_edge_triggers.py

```diff
@@ -233,3 +233,39 @@ async def wait_falling_edge():
     falling_coro = cocotb.fork(wait_falling_edge())
     cocotb.fork(Clock(dut.clk, 10, units='ns').start())
     await Combine(rising_coro, falling_coro)
+
+
+@cocotb.test()
+async def test_edge_on_vector(dut):
+    """Test that Edge() triggers on any 0/1 change in a vector"""
+
+    cocotb.fork(Clock(dut.clk, 100, "ns").start())
+
+    edge_cnt = 0
+    edge_cnt_exp = 0
+
+    async def wait_edge():
+        nonlocal edge_cnt
+        while True:
+            await Edge(dut.stream_out_data_registered)
+            edge_cnt = edge_cnt + 1
+
+    cocotb.fork(wait_edge())
+
+    dut.stream_in_data <= 0
+    await ClockCycles(dut.clk, 1)
+
+    for val in range(1, 2**len(dut.stream_in_data)-1):
+        # produce an edge by setting a value != 0:
+        dut.stream_in_data <= val
+        await ClockCycles(dut.clk, 1)
+        edge_cnt_exp = edge_cnt_exp + 1
+        # set back to all-0:
+        dut.stream_in_data <= 0
+        await ClockCycles(dut.clk, 1)
+        edge_cnt_exp = edge_cnt_exp + 1
+
+    if cocotb.SIM_NAME.lower().startswith(("modelsim")):
+        assert edge_cnt == 1
```

Comment:
Big yikes. Why is this? Maybe a comment would be good. I think it would be better to mark this as an expected fail rather than baking it into the test. It will be easier to locate issues with simulators in the future.

Reply:
Why... no idea. At least it's consistent between VPI and FLI.

Reply:
I think this is worth opening an issue about. Is it hanging on the first edge seen? What happens if you put a `ReadOnly` after every watch?

Reply:
Cool, the extra `ReadOnly` fixed it.

Reply:
So `Edge(signal)` immediately after `Edge(signal)` doesn't work on Modelsim.

Reply:
That makes no sense since there is a `RisingEdge` after `RisingEdge` in the main process...